src/derive/derive.cpp

```diff
@@ -219,6 +219,51 @@ void incflo::ComputeStrainrate(Real time_in)
     }
 }
 
+
+Real incflo::ComputeKineticEnergy()
+{
+    BL_PROFILE("incflo::ComputeKineticEnergy");
+
+    // integrated total Kinetic energy
+    Real KE = 0.0;
+
+    for(int lev = 0; lev <= finest_level; lev++)
+    {
+        Real cell_vol = geom[lev].CellSize()[0]*geom[lev].CellSize()[1]*geom[lev].CellSize()[2];
+
+        KE += amrex::ReduceSum(*density[lev],*vel[lev],*level_mask[lev],0,
+        [=] AMREX_GPU_HOST_DEVICE (Box const& bx, FArrayBox const& denfab, FArrayBox const& velfab, IArrayBox const& maskfab) -> Real
```

Comment:
It's slightly more efficient to use `(Box const& bx, Array4<Real const> const& den, Array4<Real const> const& vel, Array4<int const> const& mask)` instead of FABs.